### PR TITLE
Saliency map visualization

### PR DESCRIPTION
**Pull request by igor-davidyuk**
This PR introduces means to extract saliency maps from predictions and prepare overlays for visualization

### CODE REVIEW

File: geti_sdk/prediction_visualization/visualizer.py [hunk 1]
```diff
@@ -93,6 +94,47 @@ def draw(
...
+            )
+        # Accessing the saliency map for the label
+        saliency_map = saliency_map[label_name]
+        if saliency_map.shape[:2] != image.shape[:2]:
+            saliency_map = cv2.resize(saliency_map, (image.shape[1], image.shape[0]))
+        # Visualizing the saliency map
+        overlay = cv2.applyColorMap(saliency_map, cv2.COLORMAP_JET)
+        overlay = cv2.cvtColor(overlay, cv2.COLOR_BGR2RGB)
+        result = cv2.addWeighted(image, 1 - opacity, overlay, opacity, 0)
+        if show_predictions:
```
Comment (ljcornel) [inline geti_sdk/prediction_visualization/visualizer.py:133]: Nice option!

File: geti_sdk/deployment/deployed_model.py [hunk 2]
```diff
@@ -489,99 +489,6 @@ def _postprocess(
         """
         return self._inference_model.postprocess(inference_results, metadata)
 
-    def _postprocess_explain_outputs(
```
Comment (ljcornel) [inline geti_sdk/deployment/deployed_model.py:492]: Great that we can get rid of this, I think it was still a remnant from before the OTX models were fully switched to model API.

File: geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py [hunk 3]
```diff
@@ -131,13 +159,11 @@ class DetectionToPredictionConverter(InferenceResultsToPredictionConverter):
...
     """
 
-    domain = Domain.DETECTION
-
     def __init__(
         self, label_schema: LabelSchema, configuration: Optional[Dict[str, Any]] = None
     ):
         self.labels = label_schema.get_labels(include_empty=False)
-        self.label_map = dict(enumerate(self.labels))
+        # self.label_map = dict(enumerate(self.labels))
```
Comment (ljcornel) [inline geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py:166]: Could you remove this line completely?

File: geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py [hunk 4]
```diff
@@ -355,29 +424,45 @@ class SegmentationToPredictionConverter(InferenceResultsToPredictionConverter):
...
+
+        :param inference_results: classification labels represented in ModelAPI format (label_index, label_name, confidence)
+        :param image_shape: shape of the input image
+        :return: Prediction object with corresponding label
+        """
+        saliency_map = inference_results.saliency_map
+        if len(saliency_map) == 0:
+            return None
+        saliency_map = np.transpose(saliency_map, (2, 0, 1))  # shape: (N classes, h, w)
+        return {label.name: saliency_map[i + 1] for i, label in enumerate(self.labels)}
```
Comment (ljcornel) [inline geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py:464]: Isn't saliency map zero-indexed as well? i.e. shouldn't it just be ` saliency_map[i]` ? Or does the ` saliency_map[0]`  entry contain the saliency map for the background?

**Review by ljcornel — COMMENTED** [review]
Looks great, thanks! Few comments/questions here and there, nothing major.

File: geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py (hunk 4 above)
Comment (igor-davidyuk) [reply geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py:464]: I tried to find out how it is processed in model API, but it is not clear at all.
Empirically, yes, this is the correct way to index the map.

File: geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py (hunk 3 above)
Comment (igor-davidyuk) [reply geti_sdk/deployment/predictions_postprocessing/results_converter/results_to_prediction_converter.py:166]: removed

File: geti_sdk/prediction_visualization/visualizer.py [hunk 5]
```diff
@@ -93,6 +94,47 @@ def draw(
         )
         return result
 
+    def explain_label(
+        self,
+        image: np.ndarray,
+        prediction: Prediction,
+        label_name: str,
+        opacity: float = 0.5,
+        show_predictions: bool = True,
```
[the remark this answers is not in the document]
Comment (igor-davidyuk) [reply geti_sdk/prediction_visualization/visualizer.py:103]: added

**Pull request merged**